mscore/pianotools.cpp

```diff
@@ -189,6 +189,17 @@ void HPiano::changeSelection(Selection selection)
             key->update();
       }
 
+// used when currentScore() is NULL; same as above except the for loop
+void HPiano::clearSelection()
+      {
+      for (PianoKeyItem* key : keys) {
+            key->setHighlighted(false);
+            key->setSelected(false);
+            }
+      for (PianoKeyItem* key : keys)
```

Comment:
Same loop condition as above; merge it into a single loop? Or is there another reason (other than copy-pasted from changeSelection) to keep it separate?

Reply:
Oh sorry, it was, indeed, copy-pasted.

I initially tried something with setting `selection` to `NULL` but ran into C++ not being C, so I did that, at almost 5 o’clock in the night. I was mostly focussed on trying to get the crash fixed in the first place.

If using two separate methods is indeed the way to go, then, yes, this should definitely be merged into one.